### PR TITLE
Extract ThrottleTimeCounter for PerWorkerMetrics

### PR DESCRIPTION
**Pull request by JayajP**
Throttled time represents the amount of time waiting due to RPC retries. It is used to make better autoscaling decisions in our backend.

Currently, we report throttled time metrics for the BigQuery StreamingInserts API, with this PR we also report the throttled time for BigQuery StorageWrite API.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by JayajP** [comment]
R: @m-trieu

**Review by m-trieu — APPROVED** [review]
LGTM

File: runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/streaming/StageInfoTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import org.apache.beam.runners.dataflow.worker.StreamingStepMetricsContainer;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQuerySinkMetrics;
+import org.apache.beam.sdk.metrics.MetricsEnvironment;
+import org.junit.Test;
+
+public class StageInfoTest {
```
Comment (m-trieu) [inline runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/streaming/StageInfoTest.java:28]: nit: annotate class with `@RunWith(JUnit4.class)`

File: runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/StageInfo.java [hunk 2]
```diff
@@ -117,6 +119,24 @@ public List<PerStepNamespaceMetrics> extractPerWorkerMetricValues() {
...
+
+  private void translateKnownPerWorkerCounters(List<PerStepNamespaceMetrics> metrics) {
+    for (PerStepNamespaceMetrics perStepnamespaceMetrics : metrics) {
+      if (!BigQuerySinkMetrics.METRICS_NAMESPACE.equals(
+          perStepnamespaceMetrics.getMetricsNamespace())) {
+        continue;
+      }
+      for (MetricValue metric : perStepnamespaceMetrics.getMetricValues()) {
+        if (BigQuerySinkMetrics.THROTTLED_TIME.equals(metric.getMetric())) {
+          Long msecs = metric.getValueInt64();
```
Comment (m-trieu) [inline runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/StageInfo.java:134]: nit: use primitive `long`

**Review by m-trieu — APPROVED** [review]
some minor comments, but other than that lgtm

File: runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/StageInfo.java (hunk 2 above)
Comment (JayajP) [reply runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/StageInfo.java:134]: Leaving this as is.
`getValueInt64` returns a Long, but it's possible - though unexpected - for the function to return null.

Added a test case to cover this case.

**Comment by Abacn** [comment]
testStreamsStartCorrectly (org.apache.beam.runners.dataflow.worker.windmill.client.grpc.StreamingEngineClientTest) failed

```
org.junit.runners.model.TestTimedOutException: test timed out after 600 seconds
	at org.apache.beam.runners.dataflow.worker.windmill.client.AbstractWindmillStream.close(AbstractWindmillStream.java)
	at org.apache.beam.runners.dataflow.worker.windmill.client.grpc.StreamingEngineClient.finish(StreamingEngineClient.java:236)
	at org.apache.beam.runners.dataflow.worker.windmill.client.grpc.StreamingEngineClientTest.cleanUp(StreamingEngineClientTest.java:172)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```

I see many different tests in worker flaky (stuck within timeout). I do not remember this one before so triggered rerun. However, we should pay attention to the test health of dataflow-runner-streaming-worker. It is at a pretty bad state currently

**Comment by JayajP** [comment]
Looks like test passes on the most recent run.

**Pull request merged**